Source/ProgressionSystemRuntime/Private/Data/PSSaveGameData.cpp

```diff
@@ -119,13 +122,43 @@ void UPSSaveGameData::UnlockAllLevels()
 // Retrieves the progression reward based on the end game state for the current level.
 float UPSSaveGameData::GetProgressionReward(EEndGameState EndGameState)
 {
-	constexpr float DefaultMultiplier = 1.0f;
-	const float DifficultyMultiplier = UPSWorldSubsystem::Get().GetDifficultyMultiplier();
-	const FPSRowData& CurrentProgressionSettingsRowData = UPSWorldSubsystem::Get().GetCurrentProgressionSettingsRowByName();
+	constexpr float DefaultProgressionReward = 0.f;
 
-	const float* LevelReward = CurrentProgressionSettingsRowData.ProgressionEndGameValues.Find(EndGameState);
-	const float ProgressionReward = LevelReward ? *LevelReward : DefaultMultiplier;
-	return ProgressionReward * DifficultyMultiplier;
+	const UCurveTable* DifficultyCurveTable = UPSDataAsset::Get().GetProgressionDifficultyMultiplierCurveTable();
+	if (!ensureMsgf(DifficultyCurveTable, TEXT("ASSERT: [%i] %s:\n'DifficultyCurveTable' is not valid!"), __LINE__, *FString(__FUNCTION__)))
```

Comment:
You can ignore it now, but in future please start slowly replacing `*FString(__FUNCTION__)` with `__FUNTION__` utlizing `%hs` instead of `%s`. You still can find `*FString(__FUNCTION__)` in my code, but I'm on my way to change them all

Reply:
done: https://github.com/moinm3uw/ProgressionSystem/pull/57/commits/47ceae05d9b63c53821d6793a4121b49e500267b